SiteAddressChanger: Fix height issue in small viewports

spen opened the pull request:
### Summary

While developing the `SiteAddressChanger` I mistakenly used `flex-basis` for the width of the component. The problem is, in smaller screen sizes/viewports the `flex-direction` switches to vertical and so the `flex-basis` effects the vertical height.

<img width="613" alt="screen shot 2018-04-23 at 13 25 23" src="https://user-images.githubusercontent.com/4335450/39187397-1c4dc39a-47c5-11e8-9672-41d29051cc93.png">

This PR fixes that by replacing `flex-basis` with `width` :)

### Testing

- Go to http://calypso.localhost:3000/domains/manage/your-site.wordpress.com/edit/your-site.wordpress.com
- Emulate or resize your browser so that it's width is under `660px`
- You should not see the issue shown in the screen shot above, instead, the component should show a natural height

pablinos commented:
Good stuff, it solves that problem! 

There's an issue with the text beneath. It's stretching beyond the width of the screen and gets obscured by the help button.

<img width="386" alt="screen shot 2018-04-24 at 16 21 44" src="https://user-images.githubusercontent.com/96462/39197369-331067b0-47dc-11e8-8342-b353f9c9def2.png">

There's a `width:400px` rule set on `.simple-site-rename-form__info` which is causing it. Might be worth changing that to 100% and adding some padding to get it away from the edge, like the form above it.

As this isn't related, should it be a separate PR/issue?

spen commented:
Bizarre, I've fixed this by just removing the line altogether. I'm not sure why I added `flex-basis` in the first place but there doesn't seem to be any regression by removing.
